Tensile/KernelWriter.py

```diff
@@ -3543,14 +3558,22 @@ def initKernel(self, kernel, tensorParametersA, tensorParametersB ):
     # MergeRead 1: ds_readAx1 ds_readBx1 mfma | ds_readAx1 ds_readAx1 mfma | => ds_readAx2 ds_readBx1 ds_readBx1 mfma | mfma |
     MergeRead = 0
     if not kernel["ProblemType"]["TLUA"] or MergeRead or self.allowLRVWforTLUandMI:
-      self.lrvwA = kernel["LocalReadVectorWidth"]
+      if (not kernel["ProblemType"]["TLUA"]) and kernel["DirectToVgprA"]:
+        # DirectToVgpr + TLU=False case, ignore LocalReadVectorWidth and use GlobalLoadVectorWidth instead.
+        self.lrvwA = vwa
+      else:
+        self.lrvwA = kernel["LocalReadVectorWidth"]
     else:
       if kernel["EnableMatrixInstruction"]:
         self.lrvwA = kernel["MIInputPerThread"]
       else:
         self.lrvwA = 1
     if not kernel["ProblemType"]["TLUB"] or MergeRead or self.allowLRVWforTLUandMI:
-      self.lrvwB = kernel["LocalReadVectorWidth"]
+      if (not kernel["ProblemType"]["TLUB"]) and kernel["DirectToVgprB"]:
+        # DirectToVgpr + TLU=False case, ignore LocalReadVectorWidth and use GlobalLoadVectorWidth instead.
```

Comment:
Will this result in generating kernels with different names but the same code, if VW is overridden in KernelWriter?

Reply:
It is not kernel variable.
I do not think self.lrvwA/B is included in kernel name.